rbi/core/kernel.rbi

```diff
@@ -223,9 +223,9 @@ module Kernel
   sig do
     params(
         arg0: String,
-        arg1: Binding,
-        filename: String,
-        lineno: Integer,
+        arg1: T.nilable(Binding),
+        filename: T.nilable(String),
```

Comment:
I think your original sig was correct (if I recall correctly).

Keep in mind that the only place an _explicit_ nil is allowed is for `arg1`. eval raises an exception when explicit nils are passed for `filename` or `lineno`, but works if they are omitted entirely - sorbet also type-checks that case correctly due the presence of default values (I'm assuming). In other words, I think the sig should only be making `arg` nilable, not the rest. 

In summary, Ruby allows these:

`
eval ":a"
eval ":a", nil
eval ":a", nil, "filename.rb", 123
`

But these are not allowed by Ruby:

`
eval ":a", nil, nil, 12
eval ":a", nil, nil, nil
eval ":a", nil, 12, nil
`

Changing the PR to only allow `arg` to be nilable should allow Sorbet to match Ruby, unless I'm missing something.

Reply:
you are right, the filename can be omit, but must be string. I repush it